Library/Homebrew/download_strategy.rb

```diff
@@ -49,38 +53,57 @@ def stage
   # Location of the cached download.
   #
   # @api public
-  sig { returns(Pathname) }
+  sig { returns(T.nilable(Pathname)) }
   attr_reader :cached_location
 
-  attr_reader :cache, :meta, :name, :version
+  sig { returns(Pathname) }
+  attr_reader :cache
+
+  sig { returns(T::Hash[Symbol, T.untyped]) }
+  attr_reader :meta
+
+  sig { returns(String) }
+  attr_reader :name
+
+  sig { returns(T.any(NilClass, String, Version)) }
+  attr_reader :version
 
   private :meta, :name, :version
 
+  sig { params(url: String, name: String, version: T.any(NilClass, String, Version), meta: T.untyped).void }
   def initialize(url, name, version, **meta)
+    @cached_location = T.let(nil, T.nilable(Pathname))
+    @ref_type = T.let(nil, T.nilable(Symbol))
+    @ref = T.let(nil, T.untyped)
     @url = url
     @name = name
     @version = version
-    @cache = meta.fetch(:cache, HOMEBREW_CACHE)
-    @meta = meta
-    @quiet = false
+    @cache = T.let(meta.fetch(:cache, HOMEBREW_CACHE), Pathname)
+    @meta = T.let(meta, T::Hash[Symbol, T.untyped])
+    @quiet = T.let(false, T.nilable(T::Boolean))
     extend Pourable if meta[:bottle]
   end
 
   # Download and cache the resource at {#cached_location}.
   #
   # @api public
+  sig { overridable.params(timeout: T.any(Float, Integer, NilClass)).void }
   def fetch(timeout: nil); end
 
+  sig { returns(Pathname) }
+  def cached_location! = T.must(cached_location)
+
   # Disable any output during downloading.
   #
   # @api public
   sig { void }
   def quiet!
-    @quiet = true
+    @quiet = T.let(true, T.nilable(T::Boolean))
   end
 
+  sig { returns(T::Boolean) }
   def quiet?
-    Context.current.quiet? || @quiet
+    Context.current.quiet? || @quiet || false
   end
```

Comment:
Why nilable? It's set to `false` in the constructor.

Reply:
Good question, and I had the same reaction. I haven't had time to chase this down, I was thinking it might have to do with super/subclass interactions in calling `super` (it was failing at runtime, not statically).